### ISSUE TITLE
 JLab on Big Sur -- AssertionError: Couldn't find Class NSProcessInfo

### ISSUE THREAD

**Issue by matteosantama** [jupyterlab/jupyterlab#9410]
Jupyter Lab is unable to start a kernel after upgrading to Big Sur on Mac. 

### Steps to reproduce
1. Uninstall and re-install conda. 
2. Install JL
2. Run `jupyter lab`

### Error
```
[I 11:24:15.462 LabApp] Build is up to date
[I 11:24:17.850 LabApp] Creating new notebook in 
[I 11:24:18.108 LabApp] Kernel started: 6062fe09-e35d-489e-85fc-3c82100f9a76, name: python3
Traceback (most recent call last):
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/ipykernel_launcher.py", line 16, in <module>
    app.launch_new_instance()
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/traitlets/config/application.py", line 844, in launch_instance
    app.initialize(argv)
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/traitlets/config/application.py", line 87, in inner
    return method(app, *args, **kwargs)
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/ipykernel/kernelapp.py", line 582, in initialize
    self.init_kernel()
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/ipykernel/kernelapp.py", line 452, in init_kernel
    user_ns=self.user_ns,
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/traitlets/config/configurable.py", line 537, in instance
    inst = cls(*args, **kwargs)
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/ipykernel/ipkernel.py", line 86, in __init__
    appnope.nope()
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/appnope/_nope.py", line 90, in nope
    "Because Reasons"
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/appnope/_nope.py", line 60, in beginActivityWithOptions
    NSProcessInfo = C('NSProcessInfo')
  File "/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages/appnope/_nope.py", line 38, in C
    assert ret is not None, "Couldn't find Class %s" % classname
AssertionError: Couldn't find Class NSProcessInfo
```

### Jupyter Troubleshoot

```
$PATH:
	/Users/matteosantamaria/opt/miniconda3/envs/lotto/bin
	/Users/matteosantamaria/opt/miniconda3/condabin
	/usr/local/opt/python/libexec/bin
	/Users/matteosantamaria/Library/Python/3.7/bin
	/usr/local/bin
	/usr/bin
	/bin
	/usr/sbin
	/sbin
	/Library/TeX/texbin
	/opt/X11/bin

sys.path:
	/Users/matteosantamaria/opt/miniconda3/envs/lotto/bin
	/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python37.zip
	/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7
	/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/lib-dynload
	/Users/matteosantamaria/opt/miniconda3/envs/lotto/lib/python3.7/site-packages

sys.executable:
	/Users/matteosantamaria/opt/miniconda3/envs/lotto/bin/python

sys.version:
	3.7.8 | packaged by conda-forge | (default, Nov 27 2020, 19:11:24) 
	[Clang 11.0.0 ]

platform.platform():
	Darwin-20.1.0-x86_64-i386-64bit

which -a jupyter:
	/Users/matteosantamaria/opt/miniconda3/envs/lotto/bin/jupyter

pip list:
	Package                       Version
	----------------------------- -------------------
	appnope                       0.1.0
	argon2-cffi                   20.1.0
	async-generator               1.10
	attrs                         20.3.0
	backports.functools-lru-cache 1.6.1
	bleach                        3.2.1
	brotlipy                      0.7.0
	certifi                       2020.11.8
	cffi                          1.14.4
	chardet                       3.0.4
	cryptography                  3.2.1
	decorator                     4.4.2
	defusedxml                    0.6.0
	entrypoints                   0.3
	idna                          2.10
	importlib-metadata            3.1.0
	ipykernel                     5.3.4
	ipython                       5.8.0
	ipython-genutils              0.2.0
	Jinja2                        2.11.2
	json5                         0.9.5
	jsonschema                    3.2.0
	jupyter-client                6.1.7
	jupyter-core                  4.7.0
	jupyterlab                    2.2.9
	jupyterlab-pygments           0.1.2
	jupyterlab-server             1.2.0
	MarkupSafe                    1.1.1
	mistune                       0.8.4
	nbclient                      0.5.1
	nbconvert                     6.0.7
	nbformat                      5.0.8
	nest-asyncio                  1.4.3
	notebook                      6.1.5
	packaging                     20.4
	pandocfilters                 1.4.2
	pexpect                       4.8.0
	pickleshare                   0.7.5
	pip                           20.3
	prometheus-client             0.9.0
	prompt-toolkit                1.0.15
	ptyprocess                    0.6.0
	pycparser                     2.20
	Pygments                      2.7.2
	pyOpenSSL                     20.0.0
	pyparsing                     2.4.7
	pyrsistent                    0.17.3
	PySocks                       1.7.1
	python-dateutil               2.8.1
	pyzmq                         20.0.0
	requests                      2.25.0
	Send2Trash                    1.5.0
	setuptools                    49.6.0.post20201009
	simplegeneric                 0.8.1
	six                           1.15.0
	terminado                     0.9.1
	testpath                      0.4.4
	tornado                       6.1
	traitlets                     5.0.5
	urllib3                       1.25.11
	wcwidth                       0.2.5
	webencodings                  0.5.1
	wheel                         0.35.1
	zipp                          3.4.0

conda list:
	# packages in environment at /Users/matteosantamaria/opt/miniconda3/envs/lotto:
	#
	# Name                    Version                   Build  Channel
	appnope                   0.1.0           py37hf985489_1003    conda-forge
	argon2-cffi               20.1.0           py37h4b544eb_2    conda-forge
	async_generator           1.10                       py_0    conda-forge
	attrs                     20.3.0             pyhd3deb0d_0    conda-forge
	backports                 1.0                        py_2    conda-forge
	backports.functools_lru_cache 1.6.1                      py_0    conda-forge
	bleach                    3.2.1              pyh9f0ad1d_0    conda-forge
	brotlipy                  0.7.0           py37h395d20d_1001    conda-forge
	ca-certificates           2020.11.8            h033912b_0    conda-forge
	certifi                   2020.11.8        py37hf985489_0    conda-forge
	cffi                      1.14.4           py37hc5b2277_1    conda-forge
	chardet                   3.0.4           py37h2987424_1008    conda-forge
	cryptography              3.2.1            py37h3b7a55b_0    conda-forge
	decorator                 4.4.2                      py_0    conda-forge
	defusedxml                0.6.0                      py_0    conda-forge
	entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
	idna                      2.10               pyh9f0ad1d_0    conda-forge
	importlib-metadata        3.1.0              pyhd8ed1ab_0    conda-forge
	importlib_metadata        3.1.0                hd8ed1ab_0    conda-forge
	ipykernel                 5.3.4            py37he01cfaa_1    conda-forge
	ipython                   5.8.0                    py37_1    conda-forge
	ipython_genutils          0.2.0                      py_1    conda-forge
	jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
	json5                     0.9.5              pyh9f0ad1d_0    conda-forge
	jsonschema                3.2.0                      py_2    conda-forge
	jupyter_client            6.1.7                      py_0    conda-forge
	jupyter_core              4.7.0            py37hf985489_0    conda-forge
	jupyterlab                2.2.9                      py_0    conda-forge
	jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
	jupyterlab_server         1.2.0                      py_0    conda-forge
	libcxx                    11.0.0               h4c3b8ed_1    conda-forge
	libffi                    3.3                  h74dc148_1    conda-forge
	libsodium                 1.0.18               hbcb3906_1    conda-forge
	markupsafe                1.1.1            py37h395d20d_2    conda-forge
	mistune                   0.8.4           py37h4b544eb_1002    conda-forge
	nbclient                  0.5.1                      py_0    conda-forge
	nbconvert                 6.0.7            py37hf985489_3    conda-forge
	nbformat                  5.0.8                      py_0    conda-forge
	ncurses                   6.2                  h2e338ed_4    conda-forge
	nest-asyncio              1.4.3              pyhd8ed1ab_0    conda-forge
	notebook                  6.1.5            py37hf985489_0    conda-forge
	openssl                   1.1.1h               haf1e3a3_0    conda-forge
	packaging                 20.4               pyh9f0ad1d_0    conda-forge
	pandoc                    2.11.2               hc929b4f_0    conda-forge
	pandocfilters             1.4.2                      py_1    conda-forge
	pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
	pickleshare               0.7.5                   py_1003    conda-forge
	pip                       20.3               pyhd8ed1ab_0    conda-forge
	prometheus_client         0.9.0              pyhd3deb0d_0    conda-forge
	prompt_toolkit            1.0.15                     py_1    conda-forge
	ptyprocess                0.6.0                   py_1001    conda-forge
	pycparser                 2.20               pyh9f0ad1d_2    conda-forge
	pygments                  2.7.2                      py_0    conda-forge
	pyopenssl                 20.0.0             pyhd8ed1ab_0    conda-forge
	pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
	pyrsistent                0.17.3           py37h4b544eb_1    conda-forge
	pysocks                   1.7.1            py37h2987424_2    conda-forge
	python                    3.7.8           h4f09611_3_cpython    conda-forge
	python-dateutil           2.8.1                      py_0    conda-forge
	python_abi                3.7                     1_cp37m    conda-forge
	pyzmq                     20.0.0           py37h47fd9b3_1    conda-forge
	readline                  8.0                  h0678c8f_2    conda-forge
	requests                  2.25.0             pyhd3deb0d_0    conda-forge
	send2trash                1.5.0                      py_0    conda-forge
	setuptools                49.6.0           py37h2987424_2    conda-forge
	simplegeneric             0.8.1                      py_1    conda-forge
	six                       1.15.0             pyh9f0ad1d_0    conda-forge
	sqlite                    3.33.0               h960bd1c_1    conda-forge
	terminado                 0.9.1            py37hf985489_1    conda-forge
	testpath                  0.4.4                      py_0    conda-forge
	tk                        8.6.10               hb0a8c7a_1    conda-forge
	tornado                   6.1              py37h4b544eb_0    conda-forge
	traitlets                 5.0.5                      py_0    conda-forge
	urllib3                   1.25.11                    py_0    conda-forge
	wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
	webencodings              0.5.1                      py_1    conda-forge
	wheel                     0.35.1             pyh9f0ad1d_0    conda-forge
	xz                        5.2.5                haf1e3a3_1    conda-forge
	zeromq                    4.3.3                h74dc148_3    conda-forge
	zipp                      3.4.0                      py_0    conda-forge
	zlib                      1.2.11            h7795811_1010    conda-forge

conda env:
	name: lotto
	channels:
	  - conda-forge
	  - defaults
	dependencies:
	  - appnope=0.1.0=py37hf985489_1003
	  - argon2-cffi=20.1.0=py37h4b544eb_2
	  - async_generator=1.10=py_0
	  - attrs=20.3.0=pyhd3deb0d_0
	  - backports=1.0=py_2
	  - backports.functools_lru_cache=1.6.1=py_0
	  - bleach=3.2.1=pyh9f0ad1d_0
	  - brotlipy=0.7.0=py37h395d20d_1001
	  - ca-certificates=2020.11.8=h033912b_0
	  - certifi=2020.11.8=py37hf985489_0
	  - cffi=1.14.4=py37hc5b2277_1
	  - chardet=3.0.4=py37h2987424_1008
	  - cryptography=3.2.1=py37h3b7a55b_0
	  - decorator=4.4.2=py_0
	  - defusedxml=0.6.0=py_0
	  - entrypoints=0.3=pyhd8ed1ab_1003
	  - idna=2.10=pyh9f0ad1d_0
	  - importlib-metadata=3.1.0=pyhd8ed1ab_0
	  - importlib_metadata=3.1.0=hd8ed1ab_0
	  - ipykernel=5.3.4=py37he01cfaa_1
	  - ipython=5.8.0=py37_1
	  - ipython_genutils=0.2.0=py_1
	  - jinja2=2.11.2=pyh9f0ad1d_0
	  - json5=0.9.5=pyh9f0ad1d_0
	  - jsonschema=3.2.0=py_2
	  - jupyter_client=6.1.7=py_0
	  - jupyter_core=4.7.0=py37hf985489_0
	  - jupyterlab=2.2.9=py_0
	  - jupyterlab_pygments=0.1.2=pyh9f0ad1d_0
	  - jupyterlab_server=1.2.0=py_0
	  - libcxx=11.0.0=h4c3b8ed_1
	  - libffi=3.3=h74dc148_1
	  - libsodium=1.0.18=hbcb3906_1
	  - markupsafe=1.1.1=py37h395d20d_2
	  - mistune=0.8.4=py37h4b544eb_1002
	  - nbclient=0.5.1=py_0
	  - nbconvert=6.0.7=py37hf985489_3
	  - nbformat=5.0.8=py_0
	  - ncurses=6.2=h2e338ed_4
	  - nest-asyncio=1.4.3=pyhd8ed1ab_0
	  - notebook=6.1.5=py37hf985489_0
	  - openssl=1.1.1h=haf1e3a3_0
	  - packaging=20.4=pyh9f0ad1d_0
	  - pandoc=2.11.2=hc929b4f_0
	  - pandocfilters=1.4.2=py_1
	  - pexpect=4.8.0=pyh9f0ad1d_2
	  - pickleshare=0.7.5=py_1003
	  - pip=20.3=pyhd8ed1ab_0
	  - prometheus_client=0.9.0=pyhd3deb0d_0
	  - prompt_toolkit=1.0.15=py_1
	  - ptyprocess=0.6.0=py_1001
	  - pycparser=2.20=pyh9f0ad1d_2
	  - pygments=2.7.2=py_0
	  - pyopenssl=20.0.0=pyhd8ed1ab_0
	  - pyparsing=2.4.7=pyh9f0ad1d_0
	  - pyrsistent=0.17.3=py37h4b544eb_1
	  - pysocks=1.7.1=py37h2987424_2
	  - python=3.7.8=h4f09611_3_cpython
	  - python-dateutil=2.8.1=py_0
	  - python_abi=3.7=1_cp37m
	  - pyzmq=20.0.0=py37h47fd9b3_1
	  - readline=8.0=h0678c8f_2
	  - requests=2.25.0=pyhd3deb0d_0
	  - send2trash=1.5.0=py_0
	  - setuptools=49.6.0=py37h2987424_2
	  - simplegeneric=0.8.1=py_1
	  - six=1.15.0=pyh9f0ad1d_0
	  - sqlite=3.33.0=h960bd1c_1
	  - terminado=0.9.1=py37hf985489_1
	  - testpath=0.4.4=py_0
	  - tk=8.6.10=hb0a8c7a_1
	  - tornado=6.1=py37h4b544eb_0
	  - traitlets=5.0.5=py_0
	  - urllib3=1.25.11=py_0
	  - wcwidth=0.2.5=pyh9f0ad1d_2
	  - webencodings=0.5.1=py_1
	  - wheel=0.35.1=pyh9f0ad1d_0
	  - xz=5.2.5=haf1e3a3_1
	  - zeromq=4.3.3=h74dc148_3
	  - zipp=3.4.0=py_0
	  - zlib=1.2.11=h7795811_1010
	prefix: /Users/matteosantamaria/opt/miniconda3/envs/lotto
```

**Comment by matteosantama** [comment]
Possibly related to #9408

**Comment by matteosantama** [comment]
Weird, when I create a new conda environment specifying `python=3.8.5`, Jupyter Lab works no problem. Not sure why 3.7 doesn't seem to work, but at least I've found a workaround.

**Comment by mattijn** [comment]
Same here.
Very strange that only environments with python 3.8 works and all lower have the NSProcessInfo AssertionError

**Comment by yaromar** [comment]
Same here. Using environments with python 3.7 seems to cause a similar problem for my Jupiter notebooks

**Comment by AlexTo** [comment]
I think it only happened recently in the last few days. I've been using conda python 3.7 with Jupyter Lab on Big Sur for a while without issues. Only environments created in the last few days will have this issue.

Furthermore, if I install jupyterlab with pip instead of conda in a conda environment (python 3.7), it seems to work fine

**Comment by jasongrout** [comment]
Thanks @AlexTo, that is very interesting information.

The OP error was actually coming from appnope via ipykernel (i.e., it is not a problem with JupyterLab itself). There was recently a fix made in ipykernel to fix an error in appnope on the mac m1 platform. I wonder if somehow that is affecting things here.

To move forward:

1. Can someone confirm that this is actually a problem in ipykernel, not JupyterLab, by trying to just run ipykernel from the command line?
2. If so, can you file an issue with ipykernel and we can take the discussion there? Note that this is probably related to https://github.com/ipython/ipykernel/pull/561

**Comment by jasongrout** [comment]
Actually, digging a bit deeper, this may be related to https://github.com/minrk/appnope/pull/7.

**Comment by jasongrout** [comment]
Ah, looks like an issue was already filed: https://github.com/minrk/appnope/issues/12 (and it may not be related to the segfault on m1...)

**Comment by roy651** [comment]
> Possibly related to #9408
Just a clarification: Unlike #9408 this does not happen only on the new M1 Mac

**Comment by raeisi** [comment]
I have a macbook 2017 that has same issue

  File "/opt/anaconda3/envs/DES37/lib/python3.7/site-packages/appnope/_nope.py", line 38, in C
    assert ret is not None, "Couldn't find Class %s" % classname
AssertionError: Couldn't find Class NSProcessInfo

**Comment by jasongrout** [comment]
Let's move the conversation over to the appnope issue, which is where the problem seems to be: minrk/appnope#12

**Comment by jasongrout** [comment]
For whoever wants to help with this - please help by adding reproduction instructions, etc., at minrk/appnope#12.

**Comment by kevin-rowney** [comment]
This comment is intended to help anyone like me who stumbled across this issue while trying to figure out their once-functional Jupyter instance started crashing after an upgrade to Big Sur 11.1

Some of my environments worked, but any of them that ran ok in Python 3.7 before would just crash the kernel after upgrade.

After trying many of the helpful suggestions here, none worked for me.  I'm now rebuilding my Jupyter environments just to see if I can get a trustworthy outcome there for Big Sur.

**Comment by michaelpedota** [comment]
I have been having the same issues as mentioned above and have tried all suggested possible fixes. I haven't had any luck with any of the suggested fixes. I have tried updating python to 3.8.5 as suggested above and get the following error message: "Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source." Is there any new ideas to fixing the kernel issue associated with Mac OS Big Sur 11.1? Any new ideas would be very helpful.

**Comment by Schmidtbit** [comment]
I am having this issue with BigSur 11.2, appnope=0.1.2, python=3.8.5 and I am not getting any solutions.
- tried with python=3.7.9 and appnope=0.1.0 and problem remains.

**Comment by jasongrout** [comment]
Please read the issue linked above for configurations that people have working: minrk/appnope#12

**Comment by jasongrout** [comment]
(And if it's not working, please comment on that issue or open a new issue on that appnope repo)

**Comment by scottwsides** [comment]
For what its worth... I just upgraded to Big Sur (11.2.3) and I started seeing problems (but I think its related to the default ipywidgets got upgraded ... I started rebuilding some images that haven't been touched in a while). So I decided to upgrade to JupyterLab 3. 

The following does NOT work (which I mean... the server seems to start... but the URL in the browser is blank

`conda install -c conda-forge jupyterlab`

This DOES work

conda install -c condo-forge jupyterlab=3.0.10

So maybe the default version has bad dependencies or version on conda-forge is bad? Both of the above give 3.0.10 as the version BTW

**Comment by jasongrout** [comment]
@scottwsides - does the browser javascript console show any errors?

**Comment by jasongrout** [comment]
If you weren't seeing this error mentioned in the issue above, your issue is likely different and deserves a new issue for discussion.

**Comment by scottwsides** [comment]
> @scottwsides - does the browser javascript console show any errors?

forgive me... not absolutely sure how to check that. I'm going through my build/setup steps more carefully today and I'll update soon.

**Comment by jasongrout** [comment]
> not absolutely sure how to check that

This looks like a page that has instructions for common browsers: https://appuals.com/open-browser-console/

**Comment by klintan** [comment]
> For what its worth... I just upgraded to Big Sur (11.2.3) and I started seeing problems (but I think its related to the default ipywidgets got upgraded ... I started rebuilding some images that haven't been touched in a while). So I decided to upgrade to JupyterLab 3.
> 
> The following does NOT work (which I mean... the server seems to start... but the URL in the browser is blank
> 
> `conda install -c conda-forge jupyterlab`
> 
> This DOES work
> 
> conda install -c condo-forge jupyterlab=3.0.10
> 
> So maybe the default version has bad dependencies or version on conda-forge is bad? So 3.0.11 is bad but 3.0.10 is good?

I just started getting errors today. For both 3.0.10 and 3.0.12 on MacOS 11.2.3. 
I'm using pyenv with Python 3.7.9 and poetry for installation, tried pip install as well, no changes.

**Comment by krstp** [comment]
Right now the issue specifically applies to `3.6`... super annoying especially when one needs to work across diff virtual conda envs

**Comment by SeongchunYang** [comment]
Same issue now. Only on <code>3.6</code>. Just switched my env to <code>3.7</code> and no longer had this issue.

**Comment by yanyuteng** [comment]
Same issue. Only on 3.6, and on 3.8 is ok.

**Comment by tsdaemon** [comment]
Same issue on 3.9

**Comment by krassowski** [comment]
It looks like this was a bug in Python which was fixed in https://github.com/python/cpython/pull/27251. The patch was backported to 3.9.7, 3.8.12, 3.7.12, and 3.6.15. @tsdaemon what exact version of Python do you use?

**Comment by tsdaemon** [comment]
@krassowski I use `3.9.6`. Guess I need to update it to `3.9.7`. Also, this helped https://github.com/minrk/appnope/issues/12#issuecomment-741625144

**Comment by teake** [comment]
> It looks like this was a bug in Python which was fixed in [python/cpython#27251](https://github.com/python/cpython/pull/27251). The patch was backported to 3.9.7, 3.8.12, 3.7.12, and 3.6.15. @tsdaemon what exact version of Python do you use?

As far as I can see, the fix was not backported to 3.6.
